src/api-client.ts

```diff
@@ -52,24 +54,34 @@ export class ApiClient {
 
   async getCredentials(): Promise<Credential[]> {
     const credentialsURL = `/update_jobs/${this.params.jobId}/credentials`
-    const res: any = await this.client.get(credentialsURL, {
-      headers: {Authorization: this.params.credentialsToken}
-    })
-    if (res.status !== 200) {
-      throw new Error(`Unexpected status code: ${res.status}`)
-    }
```

Comment:
Turns out that this didn't actually handle any errors, we should fix that for the rest of the api methods as well, but the result so far is mostly the same as axios throws an error very similar to the one that we're intending to throw